pkg/controller/deployment/deployment_controller_test.go

```diff
@@ -778,18 +773,13 @@ func TestSyncDeploymentCreatesReplicaSet(t *testing.T) {
 
 	d := newDeployment(1, nil)
 	f.dStore = append(f.dStore, d)
+	f.objects = append(f.objects, d)
 
-	// expect that one ReplicaSet with zero replicas is created
-	// then is updated to 1 replica
-	rs := newReplicaSet(d, "deploymentrs-4186632231", 0)
-	updatedRS := newReplicaSet(d, "deploymentrs-4186632231", 1)
-	opt := newListOptions()
+	rs := newReplicaSet(d, "deploymentrs-4186632231", 1)
 
 	f.expectCreateRSAction(rs)
 	f.expectUpdateDeploymentAction(d)
-	f.expectUpdateRSAction(updatedRS)
-	f.expectListPodAction(rs.Namespace, opt)
-	f.expectUpdateDeploymentAction(d)
+	f.expectListPodAction(rs.Namespace, api.ListOptions{})
 
```

Comment:
Thanks, this is correct. 
The controller first syncs the deployment and then creates an rs with 1 replica immediately (`expectCreateRSAction`), update the deployment's revision (`expectUpdateDeploymentAction`), and then list pods to calculate deployment status (`expectListPodAction`).